### PR TITLE
refacto(all): use common social groove lint rules

### PR DESCRIPTION
**Pull request by douglasduteil**

### CODE REVIEW

**Comment by douglasduteil** [comment]
:warning: After auto fix, there is still some `no-console` erors :

```
$ eslint src/* --fix

/home/x/zzz/github/douglasduteil/code-du-travail-numerique/packages/code-du-travail-api/src/server/conf/elasticsearch.js
  9:1  error  Unexpected console statement  no-console

/home/x/zzz/github/douglasduteil/code-du-travail-numerique/packages/code-du-travail-api/src/server/data_sources/code_du_travail_numerique.js
  132:5  error  Unexpected console statement  no-console
  174:5  error  Unexpected console statement  no-console
  199:9  error  Unexpected console statement  no-console
  204:5  error  Unexpected console statement  no-console

/home/x/zzz/github/douglasduteil/code-du-travail-numerique/packages/code-du-travail-api/src/server/index.js
  37:3  error  Unexpected console statement  no-console

/home/x/zzz/github/douglasduteil/code-du-travail-numerique/packages/code-du-travail-api/src/server/routes/api.js
   33:5  error  Unexpected console statement  no-console
   47:5  error  Unexpected console statement  no-console
   68:5  error  Unexpected console statement  no-console
   87:5  error  Unexpected console statement  no-console
  104:5  error  Unexpected console statement  no-console

✖ 11 problems (11 errors, 0 warnings)

```

How should be correct it ?
The API should use [winston](https://github.com/winstonjs/winston) ? or similar ?

**Comment by lionelB** [comment]
I would start with bunyan wich handle log rotation
https://github.com/trentm/node-bunyan/blob/master/README.md#stream-type-rotating-file

**Comment by douglasduteil** [comment]
As you like :kissing: 
As it's just 11 lines to replace, do you want me to do it in this PR ?
Or you prefer that ignore the console log for now

**Comment by lionelB** [comment]
Thanks ! After some search, winston (and later winston-syslog) looks to be a good option too.

**Comment by douglasduteil** [comment]
<p align=center>[AFK discussion with @revolunet]</p>

**Comment by douglasduteil** [comment]
We might want to add logstash to log to elasticsearch
https://github.com/vanthome/winston-elasticsearch might help ;)

**Comment by douglasduteil** [comment]
:up: @lionelB

**Comment by lionelB** [comment]
@douglasduteil  not sure about the husky stuff after I read that https://twitter.com/garybernhardt/status/1062785983567917056... As I understand it, it will run hooks for every commit on the monorepo.

File: packages/code-du-travail-api/src/server/utils/logger.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// @ts-check
```
Comment (lionelB) [inline packages/code-du-travail-api/src/server/utils/logger.js]: do we need the ts-check comment  ?

Comment (douglasduteil) [reply packages/code-du-travail-api/src/server/utils/logger.js]: We will when we will start adding some typescript in here ;)

Comment (lionelB) [reply packages/code-du-travail-api/src/server/utils/logger.js]: so it will be the right time to add it :p

**Comment by douglasduteil** [comment]
:up: @lionelB

**Comment by lionelB** [comment]
> @douglasduteil not sure about the husky stuff after I read that https://twitter.com/garybernhardt/status/1062785983567917056... As I understand it, it will run hooks for every commit on the monorepo.

**Comment by douglasduteil** [comment]
> As I understand it, it will run hooks for every commit on the monorepo.

Yes. Combined with `lerna` and `lint-staged`, at every commit it will :
- run the precommit script of packages that has one
- the precommit script is mainly running lint-staged
- lint-staged will ask git what changes
- if it's under the glob of the package, it will run the autofix and add it to the current commit

**Comment by douglasduteil** [comment]
> not sure about the husky stuff after I read that [twitter.com/garybernhardt/status/1062785983567917056](https://twitter.com/garybernhardt/status/1062785983567917056)

I don't think we are in the same case. I value his feedback but having git hooks that automatically handle grunt task is time and git amend saving. For me it's the extension of having auto format on save activated on you ide

**Review by lionelB — APPROVED** [review]
let's go 
husky-massage

**Pull request merged**